js/base/errors.js

```diff
@@ -1,55 +1,241 @@
-'use strict';
+import { errorHierarchy } from './errorHierarchy.js'
 
-const errorHierarchy = require ('./errorHierarchy.js')
+// Commented out since I'm not sure this is mandatory anymore
```

Comment:
can we keep the dynamically generated errors?

Reply:
@frosty00 I could not make it work, but is this still necessary?

Reply:
hmm, technically it's not but i think in this case we should dynamically generate them inside of the transpiler like how we do for python and php and ts 

https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L1334
https://github.com/ccxt/ccxt/blob/master/ccxt.d.ts

Reply:
I'd suggest to make a separate PR for that

Reply:
@frosty00 So you're suggesting generating them at transpiling time to JS too? (like it's done for php/python)

My goal here is to have them completely defined at runtime to take advantage of code completion/hinting/etc and that will be necessary too when we move to a statically typed language

Reply:
yeah if we generate them at transpile time they will be defined at runtime

Reply:
@frosty00 all good then, and as you said this can be done separately